esphome/components/pylontech/pylontech.cpp

```diff
@@ -0,0 +1,174 @@
+#include "pylontech.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace pylontech {
+
+static const char *const TAG = "pylontech";
+static const int MAX_DATA_LENGTH_BYTES = 256;
+static const uint8_t ASCII_LF = 0x0A;
+
+PylontechComponent::PylontechComponent() {}
+
+void PylontechComponent::dump_config() {
+  this->check_uart_settings(115200, 1, esphome::uart::UART_CONFIG_PARITY_NONE, 8);
+  ESP_LOGCONFIG(TAG, "pylontech:");
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "Connection with pylontech failed!");
+  }
+  ESP_LOGCONFIG(TAG, " expecting %d batteries", this->max_battery_index_);
+
+  for (int bat_num = 0; bat_num < this->max_battery_index_; bat_num++) {
+    ESP_LOGCONFIG(TAG, " Battery %d", bat_num + 1);
+    if (this->voltage_[bat_num]) {
+      LOG_SENSOR("  ", "Voltage", this->voltage_[bat_num]);
+    }
```

Comment:
The LOG_SENSOR macro already does a nullptr check
```suggestion
    LOG_SENSOR("  ", "Voltage", this->voltage_[bat_num]);
```